Exception Handling/Task2/NumberParser.cs

```diff
@@ -6,7 +6,100 @@ public class NumberParser : INumberParser
     {
         public int Parse(string stringValue)
         {
-            throw new NotImplementedException();
+            CheckForExceptionsAndTrim(ref stringValue);
+            // Initialize a variables
+            int num = 0;
+            int n = stringValue.Length;
+            int startChar = SelectStartChar(stringValue);
+            bool negative = IsItNegativeValue(stringValue);
+
+            if (negative)
+            {
+                for (int i = startChar; i < n; i++)
+                {
+                    // Subtract 48 from the current digit 
+                    int tempnum = num * 10 - ((int)stringValue[i] - 48);
+                    if (num != 0 && Math.Sign(tempnum) != Math.Sign(num))
+                    {
+                        throw new OverflowException("Value is more than Int32");
+                    }
+                    num = tempnum;
+                }
+            }
+            else
+            {
+                for (int i = startChar; i < n; i++)
+                {
+                    // Subtract 48 from the current digit 
+                    int tempnum = num * 10 + ((int)stringValue[i] - 48);
```

Comment:
@IhorStetskyi, extract magic number (48 - it's '0' so you could extract it as ZeroCode variable)

Reply:
number 48 is extracted into ZeroCode variable